src/library_manager/llext_manager.c

```diff
@@ -667,13 +662,25 @@ uintptr_t llext_manager_allocate_module(const struct comp_ipc_config *ipc_config
 
 			tr_dbg(&lib_manager_tr, "%s depending on %s index %u, %u users",
 			       mctx->llext->name, mctx->llext->dependency[i]->name,
-			       dep_ctx[i]->start_idx, mctx->llext->dependency[i]->use_count);
+			       dep->start_idx, dep->n_dependent);
 
-			ret = llext_manager_load_module(dep_ctx[i]);
+			/*
+			 * Protected by the IPC serialization, but maybe we should protect the
+			 * dependent-count explicitly too. It is incremented when a new dependent
+			 * is identified. If it's non-zero, then some other modules also depend
+			 * on it and have already mapped it.
+			 */
+			if (dep->n_dependent++)
+				continue;
+
+			/* First user of this dependency, load it into SRAM */
+			ret = llext_manager_load_module(dep);
```

Comment:
do we need to decrement n_dependent on loading failure ?

Reply:
rrright, good catch! Updated.